src/components/GraphDialog.vue

```diff
@@ -99,7 +104,24 @@ export default Vue.extend({
       ],
     };
   },
-  computed: {},
+  computed: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    hasPerms(this: any) {
+      return canUpload(store.state.userInfo, this.permissions);
+    },
+  },
+  asyncComputed: {
+    permissions: {
+      async get(): Promise<WorkspacePermissionsSpec | null> {
+        try {
+          return api.getWorkspacePermissions(this.workspace);
+        } catch (error) {
+          return null;
+        }
+      },
```

Comment:
You only need `async` if you plan on calling `await` in the function
```suggestion
      get(): Promise<WorkspacePermissionsSpec | null> {
        try {
          return api.getWorkspacePermissions(this.workspace);
        } catch (error) {
          return null;
        }
      },
```

Reply:
I had copied the code from here for the async computed, and you'll see it has the same issue. Since this is in an async computed block, it requires that a promise is returned. That means that if we want to remove that async keyword, we'd have to replace 

```
        } catch (error) {
          return null;
        }
```

with

```
        } catch (error) {
          return new Promise(() => null);
        }
```

Or something similar.

Reply:
Do you think that's preferable to the current syntax?